Tests/AFResultTests.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  AFResultTests.swift
+//
+//  Copyright (c) 2014-2018 Alamofire Software Foundation (http://alamofire.org/)
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+@testable import Alamofire
+import Foundation
+import XCTest
+
+class AFResultTestCase: BaseTestCase {
+    let error = AFError.responseValidationFailed(reason: .unacceptableStatusCode(code: 404))
+
+    // MARK: - Value Tests
+
+    func testThatValuePropertyReturnsValueForSuccessCase() {
+        // Given, When
+        let result = AFResult<String>.success("success")
+
+        // Then
+        XCTAssertEqual(result.af.value ?? "", "success", "result value should match expected value")
+    }
+
+    func testThatValuePropertyReturnsNilForFailureCase() {
+        // Given, When
+        let result = AFResult<String>.failure(error)
+
+        // Then
+        XCTAssertNil(result.af.value, "result value should be nil for failure case")
+    }
+
+    // MARK: - Error Tests
+
+    func testThatErrorPropertyReturnsNilForSuccessCase() {
+        // Given, When
+        let result = AFResult<String>.success("success")
+
+        // Then
+        XCTAssertNil(result.af.error, "result error should be nil for success case")
+    }
+
+    func testThatErrorPropertyReturnsErrorForFailureCase() {
+        // Given, When
+        let result = AFResult<String>.failure(error)
+
+        // Then
+        XCTAssertNotNil(result.af.error, "result error should not be nil for failure case")
+    }
+
+    // MARK: - Initializer Tests
+
+    func testThatInitializerFromThrowingClosureStoresResultAsASuccess() {
```

Comment:
The following tests cover `Swift.Result` methods instead of anything in Alamofire. Candidates for deletion?
```
testThatInitializerFromThrowingClosureStoresResultAsASuccess
testThatInitializerFromThrowingClosureCatchesErrorAsAFailure
testThatFlatMapTransformsSuccessValue
```

Reply:
Ah thanks, I didn't look closely enough. Will fix.

Reply:
Deleted the entire set of tests.